client-s3/src/test/kotlin/app/cash/backfila/client/s3/RecordStrategyBackfillTest.kt

```diff
@@ -0,0 +1,61 @@
+package app.cash.backfila.client.s3
+
+import app.cash.backfila.client.NoParameters
+import app.cash.backfila.client.PrepareBackfillConfig
+import app.cash.backfila.client.s3.record.RecordStrategy
+import app.cash.backfila.client.s3.shim.FakeS3Service
+import app.cash.backfila.embedded.Backfila
+import app.cash.backfila.embedded.createWetRun
+import javax.inject.Inject
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import okio.BufferedSource
+import okio.ByteString
+import org.assertj.core.api.SoftAssertions
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+@MiskTest(startService = true)
+class RecordStrategyBackfillTest {
+  @Suppress("unused")
+  @MiskTestModule
+  val module = TestingModule()
+
+  @Inject lateinit var backfila: Backfila
+
+  @Inject lateinit var fakeS3: FakeS3Service
+
+  @BeforeEach
+  fun `loadFiles`() {
+    fakeS3.loadResourceDirectory("file-structure")
+  }
+
+  @Test
+  fun `backfilling with a record strategy that consumes no bytes fails`() {
+    with(SoftAssertions()) {
```

Comment:
is there a benefit of soft assertions with only one assertion in the test? seems like a lot of unneeded boilerplate